experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -493,11 +494,11 @@ static int ws_eink_spi_probe(struct spi_device *spi)
 	info->fix.xpanstep	= 0;
 	info->fix.ypanstep	= 0;
 	info->fix.ywrapstep	= 0;
-	info->fix.line_length	= dev_props->width * dev_props->bpp / 8;
+	info->fix.line_length	= roundup(dev_props->width, 8) * dev_props->bpp / 8;
 
-	info->var.xres			= dev_props->width;
+	info->var.xres			= roundup(dev_props->width, 8);
```

Comment:
I believe xres/yres are supposed to be the visible area.  So 122x250 in the case of this device.  xres_virtual/yres_virtual are supposed to be the addressable size.  So that's 128x250 in our case. Don't make any assumptions about the relationship between xres and xres_virtual other than that xres_virtual >= xres.  So rounding is the wrong solution.  Maybe some other device has an addressable size of 160, but a visible width of 122.  Also, don't assume that only the X-axis has a difference between the visible and addressable size.